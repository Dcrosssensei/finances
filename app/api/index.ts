export * from './helpers/transforClass';
export * from './operations';
export * from './validation';
