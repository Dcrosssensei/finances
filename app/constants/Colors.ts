
export const ColorsApp={
  base: '#fff',
  blue: '#41537d',
  gray:"#636262",
  lightgray: "#bfbfbf",
  softGray: '#edf0f2',
  red: '#e60023',
  yellow: '#ffd100',
  darkyellow: '#ffb900'
}