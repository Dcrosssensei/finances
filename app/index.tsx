import { createStackNavigator } from '@react-navigation/stack';
import Home from "./pages/home/Home";
import Details from './pages/details/Details';
import { RootStackParamList } from './routes/navigationTypes';
import Header from './components/header/Header';
import { RecordProvider } from './context/MainContext';

const Stack = createStackNavigator<RootStackParamList>();


export default function Index() {

  return (
    <RecordProvider>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={Home} options={{ headerShown: false }} />
        <Stack.Screen name="Details" component={Details} 
        options={ {
          header: ()=> (<Header back />)
        }}  
        />
      </Stack.Navigator>
    </RecordProvider>
  );
}
