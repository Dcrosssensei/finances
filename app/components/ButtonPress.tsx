import { View, Pressable, PressableProps, ViewStyle } from "react-native";
import { ReactNode } from "react";
import { ColorsApp } from "../constants";

interface ButtonPressProps extends Omit<PressableProps, 'children'> {
    children: (pressed: boolean) => ReactNode;
    personalize?: ViewStyle
    colorPress?: string
    colorinactive?: string
  }

const ButtonPress = ({children, personalize, colorPress, colorinactive, ...restProps}: ButtonPressProps) => {
    return (
        <Pressable
            {...restProps}
            style={personalize}>
            {({ pressed }) => (
                <View
                    style={[
                        {
                            padding: 15,
                        },
                        pressed
                            ? { backgroundColor: colorPress || ColorsApp.blue }
                            : { backgroundColor: colorinactive || ColorsApp.lightgray },
                    ]}>
                         {children(pressed)}
                </View>
            )}
        </Pressable>
    );
};

export default ButtonPress;
