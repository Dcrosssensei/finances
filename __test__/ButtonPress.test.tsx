import ButtonPress from "@/app/components/ButtonPress";
import { ColorsApp } from "@/app/constants";
import React from "react";
import { Text, View } from "react-native";
import renderer from "react-test-renderer";


describe('ButtonPress simple form', () => {
    let tree: any
    beforeAll(()=>{
        tree = renderer.create(
        <ButtonPress>
            {(press) => (
              <View >
                <Text>Soy un boton</Text>
              </View>
            )}
        </ButtonPress>
        ).toJSON();
    })
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
      });

    it('ButtonPress render text', () => {
        const textoSearch ="Soy un boton"
        const textInside = JSON.stringify(tree)
        const position = textInside.indexOf(textoSearch)        
        expect(position).toBeGreaterThan(0);
      });

    it('ButtonPress inactive color default is lightgray', () => {
        const colorDefault = tree.children[0].props.style[1].backgroundColor;
        expect(colorDefault).toStrictEqual(ColorsApp.lightgray);
      });

  });

describe('ButtonPress change color form', () => {
    let tree: any
    beforeAll(()=>{
        tree = renderer.create(
        <ButtonPress
            colorinactive={ColorsApp.yellow}
            colorPress={ColorsApp.darkyellow}
        >
            {(press) => (
              <View >
                <Text>Soy un boton</Text>
              </View>
            )}
        </ButtonPress>
        ).toJSON();
    })
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
      });

    it('ButtonPress render text', () => {
        const textoSearch ="Soy un boton"
        const textInside = JSON.stringify(tree)
        const position = textInside.indexOf(textoSearch)        
        expect(position).toBeGreaterThan(0);
      });

    it('ButtonPress inactive color change for prop', () => {
        const colorDefault = tree.children[0].props.style[1].backgroundColor;
        expect(colorDefault).toStrictEqual(ColorsApp.yellow);
      });

    it('ButtonPress isnt lightgray', () => {
        const colorDefault = tree.children[0].props.style[1].backgroundColor;
        expect(colorDefault).not.toEqual(ColorsApp.lightgray);
      });

  });
