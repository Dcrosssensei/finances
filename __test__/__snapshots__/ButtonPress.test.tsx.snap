// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPress change color form renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "padding": 15,
        },
        {
          "backgroundColor": "#ffd100",
        },
      ]
    }
  >
    <View>
      <Text>
        Soy un boton
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonPress simple form renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "padding": 15,
        },
        {
          "backgroundColor": "#bfbfbf",
        },
      ]
    }
  >
    <View>
      <Text>
        Soy un boton
      </Text>
    </View>
  </View>
</View>
`;
